src/warc2zim/icon_finder.py

```diff
@@ -0,0 +1,84 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+
+from bs4 import BeautifulSoup, Tag
+
+ZIM_ILLUSTRATION_SIZE = 48
+
+
+@dataclass(frozen=True)
+class Icon:
+    """Helper class to compare icons for warc2zim usage"""
+
+    url: str
+    width: int
+    height: int
+    icon: bytes
+    format: str | None
+
+    def __lt__(self, other):
+        return Icon._cmp_icons(self, other) < 0
+
+    def __gt__(self, other):
+        return Icon._cmp_icons(self, other) > 0
+
+    @classmethod
+    def _cmp_icons(cls, x: "Icon", y: "Icon") -> int:
```

Comment:
I think readability would be improved with variables named `a` and `b`. `x` and `y` confuses me because those are associated with geometrical axis and thus width/height.

Reply:
`x_dim` would be more clear with `a_min` for instannce

Reply:
Yep